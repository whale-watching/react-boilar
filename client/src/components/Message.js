import React from 'react';
import PropTypes from 'prop-types';

const Message = ({ message }) => {
  if (!message.consecutive) {
    return (
      <div className="message">
        <img src={message.sender.avatar} alt="sender" />
        <div className="message-content">
          <p className="sender">{message.sender.name}</p>
          <p className="text">{message.text}</p>
          <p className="time">{message.time}</p>
        </div>
      </div>
    );
  }

  return (
    <div className="message">
      <div className="message-content-consecutive">
        <p className="text">{message.text}</p>
        <p className="time">{message.time}</p>
      </div>
    </div>
  );
};

Message.propTypes = {
  message: PropTypes.shape({
    sender: PropTypes.shape.isRequired,
    text: PropTypes.string.isRequired,
    time: PropTypes.string.isRequired,
    consecutive: PropTypes.bool,
  }).isRequired,
};

export default Message;
