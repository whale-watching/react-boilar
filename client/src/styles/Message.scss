.message {
  display: flex;
  max-width: 500px;
  min-height: 100px;
}

.message img {
  max-height: 50px;
  margin: 5px;
}

.message-content {
  margin: 5px;
  padding: 10px;
  background: #fff;
  border-radius: 20px;
}

.message-content .sender {
  margin: 0 0 15px 0;
  color: $violet;
  font-weight: bold;
}

.message-content .text {
  margin: 5px 0;
}