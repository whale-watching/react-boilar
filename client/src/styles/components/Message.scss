.message {
  display: flex;
  max-width: 500px;
  height: auto;
}

.message img {
  max-height: 50px;
  margin: 5px;
}

.message-content,.message-content-consecutive {
  min-width: 100px;
  height: auto;
  margin: 5px;
  padding: 10px;
  background: #fff;
  border-radius: 20px;
}

.message-content-consecutive {
  margin-left: 65px;
}

.message-content .sender {
  margin: 0 0 12px 0;
  color: $violet;
  font-size: 14px;
  font-weight: bold;
}

.message-content .text,.message-content-consecutive .text {
  margin: 0;
}

.message-content .time,.message-content-consecutive .time {
  float: right;
  margin: 7px 0 0 0;
  color: $grey-text;
  font-size: 14px;
  font-style: italic;
}