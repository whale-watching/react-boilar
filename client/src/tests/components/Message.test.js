import React from 'react';
import { shallow } from 'enzyme';
import Message from '../../components/Message';
import message from '../data/message';
import messageConsecutive from '../data/messageConsecutive';

describe('<Message />', () => {
  it('should render Message correctly', () => {
    const wrapper = shallow(<Message message={message} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.message').children()).toHaveLength(2);
    expect(wrapper.find('.sender').text()).toEqual(message.sender.name);
    expect(wrapper.find('.text').text()).toEqual(message.text);
    expect(wrapper.find('.time').text()).toEqual(message.time);
    expect(wrapper.find('#avatar').prop('src')).toEqual(message.sender.avatar);
  });

  it('should render consecutive Message correctly', () => {
    const wrapper = shallow(<Message message={messageConsecutive} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.message').children()).toHaveLength(1);
    expect(wrapper.find('.text').text()).toEqual(messageConsecutive.text);
    expect(wrapper.find('.time').text()).toEqual(messageConsecutive.time);
  });
});
