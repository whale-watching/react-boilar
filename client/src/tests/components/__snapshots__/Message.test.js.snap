// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> should render Message correctly 1`] = `
<div
  className="message"
>
  <img
    alt="sender"
    id="avatar"
    src="/img/avatars/1.png"
  />
  <div
    className="message-content"
  >
    <p
      className="sender"
    >
      John
    </p>
    <p
      className="text"
    >
      Hello world!
    </p>
    <p
      className="time"
    >
      20:30
    </p>
  </div>
</div>
`;

exports[`<Message /> should render consecutive Message correctly 1`] = `
<div
  className="message"
>
  <div
    className="message-content-consecutive"
  >
    <p
      className="text"
    >
      How are you?
    </p>
    <p
      className="time"
    >
      20:31
    </p>
  </div>
</div>
`;
